### PR TITLE
build scripts: add quotes around some paths

### PR DESCRIPTION
**Pull request by tadeubas**
Avoid error when parent folder has spaces `./make_libsecp256k1.sh: line 31: cd: too many arguments`

### CODE REVIEW

**Review by fobdawg43684 — APPROVED** [review]

**Comment by SomberNight** [comment]
thx. Looks good. Could you also fix the same issue in `make_zbar.sh` and `make_libusb.sh`?

File: contrib/make_libsecp256k1.sh [hunk 1]
```diff
@@ -20,15 +20,15 @@ set -e
 
 . $(dirname "$0")/build_tools_util.sh || (echo "Could not source build_tools_util.sh" && exit 1)
 
-here=$(dirname $(realpath "$0" 2> /dev/null || grealpath "$0"))
+here=$(dirname "$(realpath "$0" 2> /dev/null || grealpath "$0")")
```
Comment (SomberNight) [inline contrib/make_libsecp256k1.sh:23]: I guess you also need quotes for the top-level then.
```suggestion
here="$(dirname "$(realpath "$0" 2> /dev/null || grealpath "$0")")"
```
(and same goes in the other files)

File: contrib/make_libusb.sh [hunk 2]
```diff
@@ -7,15 +7,15 @@ set -e
 
 . $(dirname "$0")/build_tools_util.sh || (echo "Could not source build_tools_util.sh" && exit 1)
 
-here=$(dirname $(realpath "$0" 2> /dev/null || grealpath "$0"))
+here=$(dirname "$(realpath "$0" 2> /dev/null || grealpath "$0")")
```
Comment (SomberNight) [inline contrib/make_libusb.sh:10]:
```suggestion
here="$(dirname "$(realpath "$0" 2> /dev/null || grealpath "$0")")"
```

File: contrib/make_zbar.sh [hunk 3]
```diff
@@ -16,15 +16,15 @@ set -e
 
 . $(dirname "$0")/build_tools_util.sh || (echo "Could not source build_tools_util.sh" && exit 1)
 
-here=$(dirname $(realpath "$0" 2> /dev/null || grealpath "$0"))
+here=$(dirname "$(realpath "$0" 2> /dev/null || grealpath "$0")")
```
Comment (SomberNight) [inline contrib/make_zbar.sh:19]:
```suggestion
here="$(dirname "$(realpath "$0" 2> /dev/null || grealpath "$0")")"
```

File: contrib/make_libsecp256k1.sh (hunk 1 above)
Comment (tadeubas) [reply contrib/make_libsecp256k1.sh]: It wasn't needed to pass the `cd "$CONTRIB"` on line 31. But even with this, it gives an error near the end (I don't know where to change the line that don't have quotes `libtool: install: /usr/bin/install -c .libs/libsecp256k1.dll.a /home/tadeubas/test folder spaces/electrum/contrib/secp256k1/dist/lib/libsecp256k1.dll.a`:

```sh
make  all-am
make[1]: Entering directory '/home/tadeubas/test folder spaces/electrum/contrib/secp256k1'
  CC       src/libsecp256k1_la-secp256k1.lo
  CC       src/libsecp256k1_precomputed_la-precomputed_ecmult.lo
  CC       src/libsecp256k1_precomputed_la-precomputed_ecmult_gen.lo
  CCLD     libsecp256k1_precomputed.la
  CCLD     libsecp256k1.la
make[1]: Leaving directory '/home/tadeubas/test folder spaces/electrum/contrib/secp256k1'
make  install-am
make[1]: Entering directory '/home/tadeubas/test folder spaces/electrum/contrib/secp256k1'
make[2]: Entering directory '/home/tadeubas/test folder spaces/electrum/contrib/secp256k1'
 /usr/bin/mkdir -p '/home/tadeubas/test folder spaces/electrum/contrib/secp256k1/dist/lib'
 /bin/bash ./libtool   --mode=install /usr/bin/install -c   libsecp256k1.la '/home/tadeubas/test folder spaces/electrum/contrib/secp256k1/dist/lib'
libtool: install: /usr/bin/install -c .libs/libsecp256k1.dll.a /home/tadeubas/test folder spaces/electrum/contrib/secp256k1/dist/lib/libsecp256k1.dll.a
/usr/bin/install: target 'spaces/electrum/contrib/secp256k1/dist/lib/libsecp256k1.dll.a' is not a directory
make[2]: *** [Makefile:892: install-libLTLIBRARIES] Error 1
make[2]: Leaving directory '/home/tadeubas/test folder spaces/electrum/contrib/secp256k1'
make[1]: *** [Makefile:1650: install-am] Error 2
make[1]: Leaving directory '/home/tadeubas/test folder spaces/electrum/contrib/secp256k1'
make: *** [Makefile:1643: install] Error 2
🗯  ERROR: Could not install secp256k1
```

**Comment by SomberNight** [comment]
thanks

File: contrib/make_libsecp256k1.sh (hunk 1 above)
Comment (SomberNight) [reply contrib/make_libsecp256k1.sh]: ah, so the script still fails... hmm

**Comment by SomberNight** [comment]
note: `make install` still fails:
https://github.com/spesmilo/electrum/blob/7b17626c4c58c0b32b26f2043d371a35a4806d23/contrib/make_libsecp256k1.sh#L63

```
make[1]: Leaving directory '/home/user/wspace/tmp/test spaces in folder name/electrum/contrib/secp256k1'
make  install-am
make[1]: Entering directory '/home/user/wspace/tmp/test spaces in folder name/electrum/contrib/secp256k1'
make[2]: Entering directory '/home/user/wspace/tmp/test spaces in folder name/electrum/contrib/secp256k1'
 /usr/bin/mkdir -p '/home/user/wspace/tmp/test spaces in folder name/electrum/contrib/secp256k1/dist/lib'
 /bin/bash ./libtool   --mode=install /usr/bin/install -c   libsecp256k1.la '/home/user/wspace/tmp/test spaces in folder name/electrum/contrib/secp256k1/dist/lib'
libtool: install: /usr/bin/install -c .libs/libsecp256k1.so.0.0.0 /home/user/wspace/tmp/test spaces in folder name/electrum/contrib/secp256k1/dist/lib/libsecp256k1.so.0.0.0
/usr/bin/install: target 'name/electrum/contrib/secp256k1/dist/lib/libsecp256k1.so.0.0.0' is not a directory
make[2]: *** [Makefile:946: install-libLTLIBRARIES] Error 1
make[2]: Leaving directory '/home/user/wspace/tmp/test spaces in folder name/electrum/contrib/secp256k1'
make[1]: *** [Makefile:1795: install-am] Error 2
make[1]: Leaving directory '/home/user/wspace/tmp/test spaces in folder name/electrum/contrib/secp256k1'
make: *** [Makefile:1788: install] Error 2
🗯 ERROR: Could not install secp256k1
```

not sure if this is a bug that can be fixed in libsecp256k1's build scripts, or it it's a bug in `make`...
but it looks out of scope for us.
maybe related: https://savannah.gnu.org/bugs/?712

**Comment by tadeubas** [comment]
TL;DR make (not just GNU make, but all POSIX-style make implementations) does not support pathnames containing whitespace

related:
https://stackoverflow.com/questions/17965806/how-do-i-handle-files-with-spaces-in-my-makefile
